aries_cloudagent/transport/outbound/queue/base.py

```diff
@@ -1,18 +1,19 @@
 """Base classes for the queue module."""
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 import asyncio
-from typing import Union
-
+from ....core.profile import Profile
 from ...error import TransportError
+from ....core.event_bus import EventBus, Event
 
 
-class BaseOutboundQueue(ABC):
+class BaseOutboundQueue(EventBus):
```

Comment:
Can you elaborate on your reasoning for making this class extend EventBus? I think external queuing systems should be consumers of the event bus, not their own event bus. Once they become a consumer of the event bus, we should treat external queuing systems as black boxes. They can do whatever they want at that point. But having multiple event buses could get really confusing.

Reply:
Sorry for the delay, but It is complex and I have been thinking about it.

This class is the base class to extend outbound queues. As we talked about yesterday. Currently there are two aproaches for Queues. One is this, which is already working for outbound scenarios and configured by the aca-py arg parameters. 
The other one is our integration via EventBus for a plugin integration later on. 

I thought It was a good idea to make more evident the duplicity of the feature and show the limitations of continue coding in the following way. However, integrating this in that way makes the current Redis queue extends from EventBus. So will be possible to reuse its code for future transport, Redis plugin.

Currently this implementation is not a consumer, it is just a producer. It drops messages in the queue.

Reply:
I think I see your point that a queue with producers and consumers is very similar to an event bus with notifiers and subscribers. However, the details of the queue implementation are not strictly relevant because our project is to simply create a common interface for all message delivery systems to use. In our case, this common interface happens to be the event bus due to the flexibility that affords.

To oversimplify the required changes just to demonstrate a point, basically all that should technically be required is that instead of `queue.enqueue_message` getting called directly from the conductor, an `OutboundMessageEvent` will be emitted. The Queue implementation will have a subscriber to the `OutboundMessageEvent` topic that performs the handling in the implementation-specific manner. For the redis queue implementation, for instance, it'll do the some encoding and then `redis.rpush`.

Likewise, instead of the `_queue_internal` call being made in the conductor, which subsequently descends down a pretty deep call stack, the exact same `OutboundMessageEvent` would be emitted. Instead of a redis queue implementation handling it (if our agent is so configured), the default message delivery "plugin" would receive the event through its own subscriber and handle the outbound event in a very different manner from the Redis Queue. but the entry point is identical.